src/python/pants/backend/go/util_rules/go_bootstrap.py

```diff
@@ -27,38 +27,25 @@ class GoBootstrap:
 async def _go_search_paths(
     env_tgt: EnvironmentTarget, golang_subsystem: GolangSubsystem, paths: Iterable[str]
 ) -> tuple[str, ...]:
-
-    resolve_standard, resolve_local = "<ASDF>" in paths, "<ASDF_LOCAL>" in paths
-
-    if resolve_standard or resolve_local:
-        # AsdfToolPathsResult is uncacheable, so only request it if absolutely necessary.
-        asdf_result = await Get(
-            AsdfToolPathsResult,
-            AsdfToolPathsRequest(
-                env_tgt=env_tgt,
-                tool_name=golang_subsystem.asdf_tool_name,
-                tool_description="Go distribution",
-                resolve_standard=resolve_standard,
-                resolve_local=resolve_local,
-                paths_option_name="[golang].go_search_paths",
-                bin_relpath=golang_subsystem.asdf_bin_relpath,
-            ),
-        )
-        asdf_standard_tool_paths = asdf_result.standard_tool_paths
-        asdf_local_tool_paths = asdf_result.local_tool_paths
-    else:
-        asdf_standard_tool_paths = ()
-        asdf_local_tool_paths = ()
-
-    special_strings = {
-        "<ASDF>": lambda: asdf_standard_tool_paths,
-        "<ASDF_LOCAL>": lambda: asdf_local_tool_paths,
+    asdf_result = await AsdfToolPathsResult.get_un_cachable_search_paths(
+        paths,
+        env_tgt=env_tgt,
+        tool_name=golang_subsystem.asdf_tool_name,
+        tool_description="Go distribution",
+        paths_option_name="[golang].go_search_paths",
+        bin_relpath=golang_subsystem.asdf_bin_relpath,
+    )
+    asdf_standard_tool_paths = asdf_result.standard_tool_paths
+    asdf_local_tool_paths = asdf_result.local_tool_paths
+    special_strings: dict[str, Callable[[], Iterable[str]]] = {
```

Comment:
I'm not sure we need to be replicating the old code's use of callables here: the only place they're used just uses the return type to extend a list. It looks like this was copied from the Python support code, and the actual cases where functions were needed disappeared.

As a bonus, you'd lose that annoying type annotation here.

Reply:
I did eye this but decided to leave it, I'll clean it up in both places :)

Reply:
FYI the type annotation was not to satisfy the value part, but the key. `AsdfPathString` as a key is not inferred to `str`, which makes `special_strings[string]` not typecheck.

I did swap to use the enumeration value instead, since I got the impression you find the annotation ugly :)